### PR TITLE
Fix possible errors in recursion from missing 'ftp' or 'logger' attributes

### PR DESCRIPTION
**Pull request by jeanconn**
Fix possible errors in recursion from missing 'ftp' or 'logger' attributes

### CODE REVIEW

**Comment by jeanconn** [comment]
This PR replaces #5

File: Ska/ftp.py [hunk 1]
```diff
@@ -180,6 +180,8 @@ def __getattr__(self, attr):
         """
         Fall through to SFTPClient methods, and fail if not found.
         """
+        if (attr == 'ftp') or (attr == 'logger'):
```
Comment (jeanconn) [inline Ska/ftp.py]: I borrowed this idea to just check for == to 'ftp' or 'logger' from:

http://nedbatchelder.com/blog/201010/surprising_getattr_recursion.html

I think it makes sense for our use cases.

**Comment by jeanconn** [comment]
I can also confirm that the test seems to show a useful error on the old code and doesn't take forever.

File: Ska/ftp.py (hunk 1 above)
Comment (taldcroft) [inline Ska/ftp.py]: Nice, good find.

**Comment by jeanconn** [comment]
@taldcroft I think the only thing remaining for this was a review from you.

**Pull request merged**